presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/HiveMetastoreClientFactory.java

```diff
@@ -49,12 +72,138 @@ public HiveMetastoreClientFactory(
+     * @param metastoreTrustStorePassword
+     * @return SSLContext
+     */
+    private static Optional<SSLContext> metastoreSslContext(boolean metastoreTlsEnabled, Optional<File> metastoreKeyStorePath, Optional<String> metastoreKeyStorePassword, Optional<File> metastoreTrustStorePath, Optional<String> metastoreTrustStorePassword)
+    {
+        if (!metastoreTlsEnabled || (!metastoreKeyStorePath.isPresent() && !metastoreTrustStorePath.isPresent())) {
+            return Optional.empty();
+        }
+
+        try {
+            KeyStore metastoreKeyStore = null;
+            KeyManager[] metastoreKeyManagers = null;
+            if (metastoreKeyStorePath.isPresent()) {
+                char[] keyManagerPassword;
+                try {
+                    // attempt to read the key store as a PEM file
+                    metastoreKeyStore = PemReader.loadKeyStore(metastoreKeyStorePath.get(), metastoreKeyStorePath.get(), metastoreKeyStorePassword);
+                    // for PEM encoded keys, the password is used to decrypt the specific key (and does not protect the keystore itself)
+                    keyManagerPassword = new char[0];
+                }
+                catch (GeneralSecurityException | IOException ignored) {
+                    keyManagerPassword = metastoreKeyStorePassword.map(String::toCharArray).orElse(null);
+
+                    metastoreKeyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+                    try (InputStream in = new FileInputStream(metastoreKeyStorePath.get())) {
+                        metastoreKeyStore.load(in, keyManagerPassword);
+                    }
+                }
+                validateKeyStoreCertificates(metastoreKeyStore);
+                final KeyManagerFactory metastoreKeyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                metastoreKeyManagerFactory.init(metastoreKeyStore, keyManagerPassword);
+                metastoreKeyManagers = metastoreKeyManagerFactory.getKeyManagers();
+            }
+
+            // load TrustStore if configured, otherwise use KeyStore
+            KeyStore metastoreTrustStore = metastoreKeyStore;
+            if (metastoreTrustStorePath.isPresent()) {
+                metastoreTrustStore = getTrustStore(metastoreTrustStorePath.get(), metastoreTrustStorePassword);
+            }
+
+            // create TrustManagerFactory
+            final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(metastoreTrustStore);
+
+            // get X509TrustManager
+            final TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+            if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
+                throw new RuntimeException("Unexpected default trust managers:" + Arrays.toString(trustManagers));
```

Comment:
This could be more clear here based on the condition here -
```suggestion
                throw new RuntimeException("Expected exactly one X509TrustManager, but found: " + Arrays.toString(trustManagers));
;
```